addons/udes_stock/tests/test_picking_type.py

```diff
@@ -0,0 +1,187 @@
+        picking = self.create_picking(self.picking_type_putaway,
+                            products_info=self.pack_4apples_info,
+                            confirm=True,
+                            assign=True)
+
+        mls = picking.move_line_ids
+        mls.write({'location_dest_id': self.test_location_01.id})
+        location = picking.get_suggested_locations(move_line_ids=mls)
+        self.assertEqual(location, self.test_location_01)
+
+    def test02_get_suggested_locations_by_product(self):
+        """ Suggested locations should be all where the product is at.
+        """
+        self.picking_type_putaway.u_drop_location_policy = 'by_products'
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=self.package_two.id)
+        self.create_quant(self.apple.id, self.test_location_02.id, 4)
+
+        self.create_quant(self.apple.id,
+                          self.picking_type_putaway.default_location_src_id.id,
+                          4, package_id=self.package_one.id)
+        picking = self.create_picking(self.picking_type_putaway,
+                            products_info=self.pack_4apples_info,
+                            confirm=True,
+                            assign=True)
+
+        locations = picking.get_suggested_locations(products=self.apple)
+        self.assertEqual(locations, self.test_locations)
+
+    def test03_get_suggested_locations_by_product_using_move_lines(self):
+        """ Suggested locations should be all where the product is at, this test
+            uses move_lin_ids as paramater instead of products.
+        """
+        self.picking_type_putaway.u_drop_location_policy = 'by_products'
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=self.package_two.id)
+        self.create_quant(self.apple.id, self.test_location_02.id, 4)
+
+        self.create_quant(self.apple.id,
+                          self.picking_type_putaway.default_location_src_id.id,
+                          4, package_id=self.package_one.id)
+        picking = self.create_picking(self.picking_type_putaway,
+                            products_info=self.pack_4apples_info,
+                            confirm=True,
+                            assign=True)
+
+        mls = picking.move_line_ids
+        locations = picking.get_suggested_locations(move_line_ids=mls)
+        self.assertEqual(locations, self.test_locations)
+
+    def test04_get_suggested_locations_by_package(self):
+        """ Suggested locations should be all where the products inside the
+            package are at.
+        """
+        self.picking_type_putaway.u_drop_location_policy = 'by_packages'
+        self.create_quant(self.apple.id, self.test_location_01.id, 4,
+                          package_id=self.package_two.id)
+        self.create_quant(self.banana.id, self.test_location_02.id, 4)
```

Comment:
Wouldn't this be a better test if this quant was in a package as well?

Reply:
I will add a note as discussed:
Note that for packages with multiple products it is suggested any location where any of the products are at.